baselines/ogb/ogb_jax/submission.py

```diff
@@ -0,0 +1,118 @@
+from typing import Iterator, List, Optional, Tuple
+
+from flax import jax_utils
+import jax
+from jax import lax
+import jax.numpy as jnp
+import optax
+
+from algorithmic_efficiency import spec
+
+
+def get_batch_size(workload_name):
+  # Return the per-device batch size.
+  batch_sizes = {'ogb_jax': 256}
+  return batch_sizes[workload_name]
+
+
+def init_optimizer_state(workload: spec.Workload,
+                         model_params: spec.ParameterContainer,
+                         model_state: spec.ModelAuxiliaryState,
+                         hyperparameters: spec.Hyperparamters,
+                         rng: spec.RandomState) -> spec.OptimizerState:
+  """Creates an Adam optimizer."""
+  del model_params
+  del model_state
+  del rng
+  params_zeros_like = jax.tree_map(lambda s: jnp.zeros(s.shape_tuple),
+                                   workload.param_shapes)
+  opt_init_fn, opt_update_fn = opt_init_fn, opt_update_fn = optax.adam(
+      learning_rate=hyperparameters.learning_rate)
+  optimizer_state = opt_init_fn(params_zeros_like)
+  return jax_utils.replicate(optimizer_state), opt_update_fn
+
+
+def train_step(workload,
+               opt_update_fn,
+               model_state,
+               optimizer_state,
+               current_param_container,
+               hyperparameters,
+               input_batch,
+               label_batch,
+               mask_batch,
+               rng):
+  del hyperparameters
+
+  def loss_fn(params):
+    logits_batch, new_model_state  = workload.model_fn(
+        params,
+        input_batch,
+        model_state,
+        spec.ForwardPassMode.TRAIN,
+        rng,
+        update_batch_norm=True)
+    loss = workload.loss_fn(label_batch, logits_batch, mask_batch)
+    mean_loss = jnp.sum(jnp.where(mask_batch, loss, 0)) / jnp.sum(mask_batch)
```

Comment:
In the current implementation `workload.loss_fn()` seems to already return the mean loss.

Reply:
Fixed.